web/modules/custom/dpl_url_proxy/src/Form/ProxyUrlConfigurationForm.php

```diff
@@ -0,0 +1,303 @@
+<?php
+
+namespace Drupal\dpl_url_proxy\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Messenger\MessengerTrait;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\dpl_url_proxy\DplUrlProxyInterface;
+use function Safe\sprintf;
+
+/**
+ * The administration form for handling the configuration of the DPL URL Proxy.
+ *
+ * @package Drupal\dpl_url_proxy\Form
+ */
+class ProxyUrlConfigurationForm extends ConfigFormBase {
+  use StringTranslationTrait;
+  use MessengerTrait;
+
+  public const CONFIG_NAME = 'dpl_url_proxy.settings';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      self::CONFIG_NAME,
+    ];
+  }
+
+  /**
+   * Get the url proxy configuration.
+   *
+   * @return mixed[]
+   *   The url proxy configuration.
+   */
+  protected function getConfiguration() {
+    $config = $this->config(self::CONFIG_NAME);
+    return $config->get('values') ?? [
+      'prefix' => '',
+      'hostnames' => [],
+    ];
+  }
+
+  // Phpstan and phpcs is conflicting about the return type.
+  // phpcs:disable Drupal.Commenting.FunctionComment.InvalidReturn
+
+  /**
+   * Create indexes used for the host names add more/delete part.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Drupal form state.
+   * @param mixed[] $values
+   *   Saved configuration.
+   *
+   * @return array<int|string>
```

Comment:
Consider if we can replace `array<int|string>` with `mixed[]' to fix the problem with invalid return types in the PHPCS sniff. [It seems to be a problem there](https://www.drupal.org/project/coder/issues/3303625).

An alternative solution would be to drop the Drupal return type definition sniff entirely. I for one prefer actual return type definitions or PHPStan ones anyway :smile:

Reply:
I forsee a future PR with removal of the Drupal return type definition sniff